src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -352,8 +352,9 @@ def resourceOffers(self, driver, offers):
 
         # Right now, gives priority to largest jobs
         for offer in offers:
-            tasks = []
             # TODO: In an offer, can there ever be more than one resource with the same name?
+            allTasks = []
+            subTasks = defaultdict(list)
```

Comment:
In fact I don't see the need for a dictionary here. You loop over the job types once, which means that you set each entry once and in the same iteration you access that entry. You might as well use a simple list that get overwritten in each iteration. Correct me if I am wrong.